File: problems/nailong-product-challenge/std.cpp
#include <bits/stdc++.h>
using namespace std;

int main() {
    ios::sync_with_stdio(false);
    cin.tie(nullptr);

    const long long LIMIT = 1000000000LL; // 10^9
    long long product = 1;

    for (int i = 0; i < 10; i++) {
        long long x;
        cin >> x;
        product *= x;

        // Early termination if product exceeds limit
        if (product > LIMIT) {
            cout << -1 << endl;
            return 0;
        }
    }

    cout << product << endl;

    return 0;
}

File: problems/nailong-product-challenge/std.py
import sys

def main():
    LIMIT = 10**9
    product = 1

    # Read 10 integers
    numbers = list(map(int, sys.stdin.readline().split()))

    # Ensure we have exactly 10 numbers
    if len(numbers) != 10:
        print("Error: Expected exactly 10 numbers")
        return

    for x in numbers:
        product *= x

        # Early termination if product exceeds limit
        if product > LIMIT:
            print(-1)
            return

    print(product)

if __name__ == "__main__":
    main()